content/oci/oci_test.go

```diff
@@ -452,6 +452,86 @@ func TestStore_ContentBadPush(t *testing.T) {
 	}
 }
 
+func TestStore_ResolveByTagReturnsFullDescriptor(t *testing.T) {
+	content := []byte("hello world")
+	ref := "hello-world:0.0.1"
+	annotations := map[string]string{"name": "Hello"}
+	desc := ocispec.Descriptor{
+		MediaType:   "test",
+		Digest:      digest.FromBytes(content),
+		Size:        int64(len(content)),
+		Annotations: annotations,
+	}
+
+	tempDir := t.TempDir()
+	s, err := New(tempDir)
+	if err != nil {
+		t.Fatal("New() error =", err)
+	}
+	ctx := context.Background()
+
+	err = s.Push(ctx, desc, bytes.NewReader(content))
+	if err != nil {
+		t.Errorf("Store.Push() error = %v, wantErr %v", err, false)
+	}
+
+	err = s.Tag(ctx, desc, ref)
+	if err != nil {
+		t.Errorf("error tagging descriptor error = %v, wantErr %v", err, false)
+	}
+
+	resolvedDescr, err := s.Resolve(ctx, ref)
+	if err != nil {
+		t.Errorf("error resolving descriptor error = %v, wantErr %v", err, false)
+	}
+
+	if !reflect.DeepEqual(resolvedDescr, desc) {
+		t.Errorf("Store.Resolve() = %v, want %v", resolvedDescr, desc)
+	}
+
+	// descriptor resolved by tag should have annotations
+	if !reflect.DeepEqual(resolvedDescr.Annotations, annotations) {
```

Comment:
Annotations should have been checked by `reflect.DeepEqual()`

Reply:
I removed the check as it's indeed redundant